includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -768,6 +766,105 @@ public function sanitize() {
 		}
 	}
 
+	/**
+	 * Get the priority of the stylesheet associated with the given element.
+	 *
+	 * As with hooks, lower priorities mean they should be included first.
+	 * The higher the priority value, the more likely it will be that the
+	 * stylesheet will be among those excluded due to 'excessive_css' when
+	 * concatenated CSS reaches 50KB.
+	 *
+	 * @todo This will eventually need to be abstracted to not be CMS-specific, allowing for the prioritization scheme to be defined by configuration.
+	 *
+	 * @param DOMNode|DOMElement|DOMAttr $node Node.
+	 * @return int Priority.
+	 */
+	private function get_stylesheet_priority( DOMNode $node ) {
```

Comment:
@swissspidy getting your gut feeling about the prioritization order in this method would be helpful.

Reply:
The logic in general seems to make sense, and the order is reasonable I think.

I was wondering if inline `style=""` attributes should get a higher priority (like they do in the browser). Just think about Gutenberg blocks with custom background colors and stuff. Those shouldn't be stripped out.

As for the plugin/theme detection, I remember that quite some logic was needed in `load_script_textdomain()` for a similar issue because of CDNs, different domains, etc.

Reply:
I think the theme/plugin detection will work as-is here because it's using `plugins_url()`, `get_template_directory_uri()`/`get_stylesheet_directory_uri()`. We don't need as much granularity here.

Reply:
> I was wondering if inline `style=""` attributes should get a higher priority (like they do in the browser). Just think about Gutenberg blocks with custom background colors and stuff. Those shouldn't be stripped out.

That's a good question. I think the `link` stylesheets (e.g. `wp-block-library`) should for sure be higher priority because they have the defaults which the `style` attributes in general customize. But otherwise should a `style` element have more priority than `style` attribute?  Would those styles be more important than Additional CSS?

Reply:
So right now Additional CSS has priority of 60, and style elements and attributes both have one of 70.

I'd assume that all the CSS from `style` attributes is usually way less than from `style` elements, so making the attributes have a bit of a higher priority probably wouldn't cause any problems.

But perhaps the current solution is good enough for now. 🤷‍♂

Reply:
Yeah, let's get this out for testing and we can see some real world results.